module-4/backend/week_one.md

```diff
@@ -5,15 +5,41 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's the most useful thing you learned from completing the intermission week work?
+I really like learning about how front end and back end programs are completely seperate programs and connect through APIs.  Maybe that isn't as sexy as learning AJAX or something but to me this changes the way I look at programming and opens up a lot of ideas for my personal project (like a golang backend and a react frontend)
+
 2. What are some tools to help debug JavaScript code?
+I find debugging in JS much harder than Ruby.  The error codes are not as helpful!  I use debugger, the console and the error messages in the npm start webpack server compiler to help debug.  This is an area that I feel I could def improve on so I will keep working on it.  
+
 3. What are some tools you need in order to unit test your JavaScript?
+In class we used Mocha and Chai.  Mocha is for unit testing and Chai adds a bunch of useful methods like deepEqual that normally wouldn't say two strings or two arrays are equal because they are in different places in memory.
+
 4. What is the syntax for invoking a function?
+function multiply() {
+statements
+};
+let functionName = () = {
+statements
+};
+var multiply = new Function(x, y, statements);
+
 5. What's `this` in JavaScript?
+This references the object you are inside (if you are not inside an object it will reference global or window).  The one caviot here is if you are in an object of an object it will reference the global scope.  Many consider this a bug but it was created intentionally by the high council of Javascript.  
+
 6. What is Webpack and why is it useful?
+Similar to the asset pipeline in Rails, Webpack condenses all of the assets (JS, CSS, SCSS) into one file of readable code (JS instead of JSX or CSS instead of SCSS).  This makes everything faster and easier for the user, the server and the programmer.  This does mean everything needs to be formatted correctly to complie correctly so it can causes additional issues for the programmer.  
+
 7. When/why do you want to use event delegation?
+This was the only question I had to look up.  Event delegation allows us to attach a single event listener, to a parent element, that will fire for all descendants matching a selector, whether those descendants exist now or are added in the future.  I.E. this allows us to use parents and children in our events, saving time and hassle and keeping our code DRY.
```

Comment:
yup! it's also helpful for when we want to bind events to elements that aren't on the page on document ready